test/test_autograd.py

```diff
@@ -347,6 +347,42 @@ def test_custom_function_exception(self):
         with self.assertRaisesRegex(Exception, "Simulate error on backward pass"):
             t3.sum().backward()
 
+    def test_custom_function_raises_if_input_returned_as_is_and_saved(self):
+        for save_inputs, mark_dirty, requires_grad, save_for_fw in product(*([(True, False)] * 4)):
```

Comment:
why not use parametrize? it's not a functorch-only thing.

Also, nit: the test here and the test in test_eager_transforms are similar enough that it might be worth just shoving all of them in the same place, but up to you.